ui/apps/platform/src/Containers/Violations/Details/DeploymentDetails.tsx

```diff
@@ -70,41 +72,42 @@ const compareNetworkPolicies = (a: NetworkPolicy, b: NetworkPolicy): number => {
     return a.name.localeCompare(b.name);
 };
 
-const DeploymentDetails = ({ deployment }) => {
-    // attempt to fetch related deployment to selected alert
-    const [relatedDeployment, setRelatedDeployment] = useState(deployment);
-    const [namespacePolicies, setNamespacePolicies] = useState([]);
+export type DeploymentDetailsProps = {
+    alertDeployment: NonNullable<AlertViolation['deployment']>;
+};
 
-    useEffect(() => {
-        fetchDeployment(deployment.id).then(
-            (dep) => setRelatedDeployment(dep),
-            () => setRelatedDeployment(null)
-        );
-    }, [deployment.id, setRelatedDeployment]);
+const DeploymentDetails = ({ alertDeployment }: DeploymentDetailsProps) => {
+    const [namespacePolicies, setNamespacePolicies] = useState<NetworkPolicy[]>([]);
+
+    // attempt to fetch related deployment to selected alert
+    const { deployment: relatedDeployment, error: relatedDeploymentFetchError } =
+        useFetchDeployment(alertDeployment.id);
 
     useEffect(() => {
-        fetchNetworkPoliciesInNamespace(deployment.clusterId, deployment.namespace).then(
-            (policies) => setNamespacePolicies(policies.response.networkPolicies),
+        fetchNetworkPoliciesInNamespace(alertDeployment.clusterId, alertDeployment.namespace).then(
+            // TODO Infer type from response once NetworkService.js is typed
+            (policies: NetworkPolicy[]) => setNamespacePolicies(policies ?? []),
             () => setNamespacePolicies([])
         );
-    }, [deployment.namespace, deployment.clusterId, setNamespacePolicies]);
+    }, [alertDeployment.namespace, alertDeployment.clusterId, setNamespacePolicies]);
 
-    const deploymentObj = relatedDeployment || deployment;
-    const namespacePoliciesList = namespacePolicies || [];
+    const relatedDeploymentPorts = relatedDeployment?.ports || [];
 
     return (
         <Flex
             direction={{ default: 'column' }}
             flex={{ default: 'flex_1' }}
             data-testid="deployment-details"
         >
-            {!relatedDeployment && (
+            {!relatedDeployment && relatedDeploymentFetchError && (
                 <Alert
                     variant="warning"
                     isInline
-                    title="This data is a snapshot of a deployment that no longer exists."
+                    title="There was an error fetching the deployment details. This deployment may no longer exist."
```

Comment:
Because of your excellent example to test by visible text of accessibility attributes, I searched to find no use of `data-testid="deployment-snapshot-warning"`

Possible follow up to rewrite 2 tests not to depend on `data-testid` in this file:
https://github.com/stackrox/stackrox/blob/master/ui/apps/platform/cypress/integration/violations/violations.test.js#L122-L144